### PR TITLE
Refactor how the parser looks for sub-modules

### PR DESCRIPTION
**Pull request by nrc**
This makes the functionality usable from outside the parser

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -297,6 +297,24 @@ fn is_plain_ident_or_underscore(t: &token::Token) -> bool {
     t.is_plain_ident() || *t == token::Underscore
 }
 
+/// Information about the path to a module.
+pub struct ModulePath {
+    pub name: String,
+    pub path_exists: bool,
```
Comment (sfackler) [inline src/libsyntax/parse/parser.rs]: This probably makes more sense inside of `ModulePathError`, right?

Comment (nrc) [inline src/libsyntax/parse/parser.rs]: It's needed in both cases. In the non-error case, there can still be an error if `!self.owns_directory` and then we need to check `path_exists`. That can't be checked without a parser, so we need to be pass out `path_exists` here.

**Comment by nrc** [comment]
@sfackler review ping

**Comment by sfackler** [comment]
@bors r+

sorry

**Pull request merged**